### ISSUE TITLE
[Bug]: NaN handling in regrid2

### ISSUE THREAD

**Issue by lee1043** [xCDAT/xcdat#580]
### What happened?

When NaN value exists in the original field, interpolation using regrid2 replaces NaN values to zero, then interpolates, which introduces error in the resulted field. `unmapped_to_nan=True` that was discussed in #528 for `xesmf` is not available for `regrid2`, wondering if that could be added. 

### What did you expect to happen? Are there are possible answers you came across?

![output](https://github.com/xCDAT/xcdat/assets/17091564/d16157bd-877a-4b35-986e-f17ae94fedaf)

Original field has NaN values over the ocean, and when interpolate, it should look like one in the middle (example used xesmf). However, regrid2 injects zeros to NaN values before interpolation, then interpolates, resulting zero over land and gradation values along coastlines. 

Minimal example code to reproduce the above plot is attached below. 

### Minimal Complete Verifiable Example (MVCE)

```python
import matplotlib.pyplot as plt
import xcdat as xc

# Generate a sample data field that has value only over ocean
input_data = "ts_Amon_ACCESS1-0_historical_r1i1p1_185001-200512.nc"
input_data2 = "sftlf_fx_ACCESS1-0_amip_r0i0p0.nc"

ds = xc.open_dataset(input_data)
ds2 = xc.open_dataset(input_data2)
sftlf = ds2["sftlf"]
ds["sst"] = ds["ts"].where(sftlf==0)

# regrid
ds_regridded_xesmf = ds.regridder.horizontal("sst", target_grid, tool="xesmf", method="bilinear")
ds_regridded_regrid2 = ds.regridder.horizontal("sst", target_grid, tool="regrid2") # unmapped_to_nan=True is not available for regrid2

fig, axs = plt.subplots(1, 3, figsize=(15, 4))

ds["sst"].isel(time=0).plot(ax=axs[0])
ds_regridded_xesmf["sst"].isel(time=0).plot(ax=axs[1])
ds_regridded_regrid2["sst"].isel(time=0).plot(ax=axs[2])

axs[0].set_title("Original")
axs[1].set_title("Regridded: xesmf")
axs[2].set_title("Regridded: regrid2")

fig.tight_layout()
```

### Anything else we need to know?

Sample input files used above:
- [ts_Amon_ACCESS1-0_historical_r1i1p1_185001-200512.nc](https://drive.google.com/file/d/1SZ5nrEI3uiHdzTi_T9iPyTKLfuvVvaIS/view?usp=sharing)
- [sftlf_fx_ACCESS1-0_amip_r0i0p0.nc](https://drive.google.com/file/d/1zV6YyKC_C1eQS5AgvxIN2ieLMor6PF1o/view?usp=sharing)

### Environment

xcdat 0.6.1

<details>
  <summary>
xr.show_versions()
</summary>

INSTALLED VERSIONS
------------------
commit: None
python: 3.10.10 | packaged by conda-forge | (main, Mar 24 2023, 20:17:34) [Clang 14.0.6 ]
python-bits: 64
OS: Darwin
OS-release: 22.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: None
LOCALE: (None, 'UTF-8')
libhdf5: 1.14.2
libnetcdf: 4.9.2

xarray: 2023.11.0
pandas: 2.1.3
numpy: 1.23.5
scipy: 1.11.4
netCDF4: 1.6.5
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.6.3
nc_time_axis: 1.4.1
iris: None
bottleneck: None
dask: 2023.12.0
distributed: 2023.12.0
matplotlib: 3.7.1
cartopy: 0.22.0
seaborn: 0.12.2
numbagg: None
fsspec: 2023.12.0
cupy: None
pint: None
sparse: 0.14.0
flox: None
numpy_groupies: None
setuptools: 67.7.2
pip: 23.1.2
conda: None
pytest: None
mypy: None
IPython: 8.18.1
sphinx: None

</details>

**Comment by lee1043** [comment]
Hi @jasonb5, is there any chance this could be fixed in #533?

**Comment by jasonb5** [comment]
@lee1043 This will be resolved in another PR after #533, it needs to be fixed for all the regridding tools.

**Comment by lee1043** [comment]
Close this issue as it was resolved by #533.

**Comment by tomvothecoder** [comment]
> Close this issue as it was resolved by #533.
> 
> Result of the minimal example code above after updating the xcdat to the latest main branch. ![output](https://private-user-images.githubusercontent.com/17091564/311936462-d626af49-f5c7-4e77-9a93-e414a4970b8b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyNjIxNDAsIm5iZiI6MTcxMDI2MTg0MCwicGF0aCI6Ii8xNzA5MTU2NC8zMTE5MzY0NjItZDYyNmFmNDktZjVjNy00ZTc3LTlhOTMtZTQxNGE0OTcwYjhiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEyVDE2NDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4ZDgyMDEwNzdiZDlhNTdjNmU4NWIxNWI5YTY0NTQ5YzkxZWNlNjI4N2QyMjdjMjAwOTkxMzViODJmOTQxNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.beTtM8dlWzlLpHjHJ5TOeJ5Up3J3YTMLzGmiC0PTbEU)

Thank you for confirming!
